### ISSUE TITLE
Access BrowserWindow as NSWindow instance from native module

### ISSUE THREAD

**Issue by callodacity** [electron/electron#10529]
Just a question:

Note that I'm new to native modules - I'm currently attempting to develop a native addon for macOS (Objective C++), and I need to be able to get the `NSWindow` instance for the given window. Meaning, I would like to design a function that would work like this:

```js
// renderer.js

const { remote } = require('electron');
const module = require('my-module');

module.showCustomDialog(remote.getCurrentWindow(), opts, callback);
```

```C++
// my-module.mm

void showCustomDialog(const FunctionCallbackInfo<Value>& args) {
  Isolate *isolate = args.GetIsolate();
  // --> Somehow get the passed window as an NSWindow here <--
}
```

I know I can get the active window with something like `[NSApplication mainWindow]` but I would like to be able to get the chosen window from the JS side of things (since I can't seem to link electron's `BrowserWindow` instances with `NSWindow` instances.

Is this currently possible?

**Comment by MarshallOfSound** [comment]
TLDR: Use https://electron.atom.io/docs/api/browser-window/#wingetnativewindowhandle 👍 

I replied to your question in Slack, we can continue the conversation there 😄

**Comment by MarshallOfSound** [comment]
GitHub issues are for feature requests and bug reports, questions about using Electron or code assistance requests should be directed to the [community](https://github.com/electron/electron#community) or to the [Slack Channel](http://atom-slack.herokuapp.com/).
